### PR TITLE
Add conformance test implementation for C# SDK

### LINKED ISSUE #2803: [Csharp SDK] Conformance test implementation
**Is your feature request related to a problem? Please describe.**

There is [unit tests for csharp](https://github.com/googleforgames/agones/blob/main/build/build-sdk-images/csharp/test.sh), but not conformance tests. 😞 

**Describe the solution you'd like**

Have a conformance test!

Will need to:

* Write an `sdktest.sh` in https://github.com/googleforgames/agones/tree/main/build/build-sdk-images/csharp that calls some C# code that calls the SDK in a way that conforms to how we expect requests to be sent (example below is the best reference).
* Add to [sdk.mk](https://github.com/googleforgames/agones/blob/main/build/includes/sdk.mk) a `run-sdk-conformance-test-csharp` target, and add it to `run-sdk-conformance-tests` target.

Examples:
* https://github.com/googleforgames/agones/blob/main/build/build-sdk-images/go/sdktest.sh
* https://github.com/googleforgames/agones/blob/main/test/sdk/go/sdk-client-test.go (which is called by above)

**Describe alternatives you've considered**

N/A - we should really have this.

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by yoshd**
**What type of PR is this?**

/kind feature

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #2803 

**Special notes for your reviewer**:

### CODE REVIEW

File: build/includes/sdk.mk [hunk 1]
```diff
@@ -180,6 +180,16 @@ run-sdk-conformance-test-rust:
...
 	DOCKER_RUN_ARGS="$(DOCKER_RUN_ARGS) -e RUN_ASYNC=true" $(MAKE) run-sdk-conformance-test SDK_FOLDER=rust GRPC_PORT=9004 HTTP_PORT=9104 FEATURE_GATES=PlayerTracking=true TESTS=$(DEFAULT_CONFORMANCE_TESTS),$(ALPHA_CONFORMANCE_TESTS)
 
+run-sdk-conformance-test-csharp:
+	# run without feature flags
+	$(MAKE) run-sdk-conformance-test SDK_FOLDER=csharp GRPC_PORT=9005 HTTP_PORT=9105
+	# run without feature flags and with RUN_ASYNC=true
+	DOCKER_RUN_ARGS="$(DOCKER_RUN_ARGS) -e RUN_ASYNC=true" $(MAKE) run-sdk-conformance-test SDK_FOLDER=csharp GRPC_PORT=9005 HTTP_PORT=9105
+	# run with feature flags enabled
+	$(MAKE) run-sdk-conformance-test SDK_FOLDER=csharp GRPC_PORT=9005 HTTP_PORT=9105 FEATURE_GATES=PlayerTracking=true TESTS=$(DEFAULT_CONFORMANCE_TESTS),$(ALPHA_CONFORMANCE_TESTS)
+	# run with feature flags enabled and with RUN_ASYNC=true
```
Comment (markmandel) [inline build/includes/sdk.mk:190]: You don't need `RUN_ASYNC` -- it was Rust specific.

You can likely copy from `run-sdk-conformance-test-go` instead.

File: test/sdk/csharp/Program.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
+﻿using Agones;
```
Comment (markmandel) [inline test/sdk/csharp/Program.cs:1]: Apache licence please 😄

**Review by markmandel — COMMENTED** [review]
You know - i swore we had one of these! Thanks for writing this!

File: test/sdk/csharp/Program.cs (hunk 2 above)
Comment (yoshd) [reply test/sdk/csharp/Program.cs]: Thank you. Fixed [2dfe70a](https://github.com/googleforgames/agones/pull/3392/commits/2dfe70a8d35692481a52ae627bfea804e8ce86a1)

**Comment by markmandel** [comment]
Found where the build was failing

```
Step #20 - "tests": Starting test execution, please wait...
Step #20 - "tests": A total of 1 test files matched the specified pattern.
Step #20 - "tests": Testhost process for source(s) '/go/src/agones.dev/agones/sdks/csharp/test/bin/Release/netcoreapp3.1/Agones.Test.dll' exited with error: You must install or update .NET to run this application.
Step #20 - "tests": App: /root/.nuget/packages/microsoft.testplatform.testhost/15.9.0/lib/netstandard1.5/testhost.dll
Step #20 - "tests": Architecture: x64
Step #20 - "tests": Framework: 'Microsoft.NETCore.App', version '3.1.0' (x64)
Step #20 - "tests": .NET location: /usr/share/dotnet/
Step #20 - "tests": The following frameworks were found:
Step #20 - "tests":   7.0.10 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]
Step #20 - "tests": Learn about framework resolution:
Step #20 - "tests": https://aka.ms/dotnet/app-launch-failed
Step #20 - "tests": To install missing framework, download:
Step #20 - "tests": https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=3.1.0&arch=x64&rid=debian.11-x64
Step #20 - "tests": . Please check the diagnostic logs for more information.
Step #20 - "tests": 
Step #20 - "tests": Test Run Aborted.
Step #20 - "tests": make[1]: *** [includes/sdk.mk:88: run-sdk-command] Error 1
Step #20 - "tests": make: *** [includes/sdk.mk:84: run-sdk-command-csharp] Error 2
```

**Comment by yoshd** [comment]
> Found where the build was failing

Thank you for finding that.
It seems to be solved by installing both dotnet-sdk-7.0 and dotnet-sdk-3.1.
(This will be resolved in #3397)

**Review by markmandel — APPROVED** [review]
Thanks!

**Pull request merged**